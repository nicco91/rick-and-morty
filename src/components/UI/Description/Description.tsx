import classNames from 'classnames';
import React, { VFC } from 'react';
import styled from 'styled-components';
import Text from '../Text';

const DescriptionWrapper = styled.div`
  display: flex;
  align-items: center;

  > :first-child {
    margin-right: 8px;
  }
`;

type Props = {
  label: string;
  content: React.ReactNode;
  className?: string;
};

const Description: VFC<Props> = ({ label, content, className }) => {
  return (
    <DescriptionWrapper className={classNames('Description', className)}>
      <Text bold>{label}:</Text>
      <div>{content}</div>
    </DescriptionWrapper>
  );
};

export default Description;
