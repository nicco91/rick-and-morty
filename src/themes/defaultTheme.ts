import { DefaultTheme } from 'styled-components';

export const defaultTheme: DefaultTheme = {
  colors: {
    primary: '#0066cc',
    success: '#00cc66',
    error: '#cc0000',
    textPrimary: '#333',
    textSecondary: '#999',
  },
  breakpoints: {
    mobile: 320,
    mobileLandscape: 480,
    tablet: 768,
    tabletLandscape: 1024,
    desktop: 1200,
    desktopLarge: 1500,
    desktopWide: 1920,
  },
};
