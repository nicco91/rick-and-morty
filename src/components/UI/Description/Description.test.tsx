import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from 'themes';
import Description from './Description';

it('render a description component', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <Description label="Sample title" content="Sample description" />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
