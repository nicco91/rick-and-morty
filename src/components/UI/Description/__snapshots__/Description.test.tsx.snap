// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a description component 1`] = `
.c1 {
  color: #333;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  font-style: normal;
  display: inline-block;
  margin: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > :first-child {
  margin-right: 8px;
}

<div
  className="c0 Description"
>
  <p
    className="c1"
  >
    Sample title
    :
  </p>
  <div>
    Sample description
  </div>
</div>
`;
